compiler/rustc_trait_selection/src/traits/project.rs

```diff
@@ -611,8 +616,43 @@ impl<'a, 'b, 'tcx> TypeFolder<TyCtxt<'tcx>> for AssocTypeNormalizer<'a, 'b, 'tcx
                 );
                 normalized_ty
             }
-
-            _ => ty.super_fold_with(self),
+            ty::Weak => {
+                let infcx = self.selcx.infcx;
+                let (data, maps) = if data.has_escaping_bound_vars() {
+                    let (data, mapped_regions, mapped_types, mapped_consts) =
+                        BoundVarReplacer::replace_bound_vars(infcx, &mut self.universes, data);
+                    (data, Some((mapped_regions, mapped_types, mapped_consts)))
+                } else {
+                    (data, None)
+                };
```

Comment:
we could only replace bound vars in the nested obligations and use the escaping bound vars in `subst`. That saves us from having to map them back to bound vars later.